packages/dev/core/src/Collisions/gpuPicker.ts

```diff
@@ -185,28 +204,31 @@ export class GPUPicker {
             const r = (id & 0xff0000) >> 16;
             const g = (id & 0x00ff00) >> 8;
             const b = (id & 0x0000ff) >> 0;
-            this._idMap[id] = index;
-            id++;
 
             if (mesh.hasThinInstances) {
-                const colorData = this._generateColorData((mesh as Mesh).thinInstanceCount, id, index, r, g, b, (i, id) => {
+                const colorData = this._generateThinInstanceColorData((mesh as Mesh).thinInstanceCount, id, index, r, g, b, (i, id) => {

```

Comment:
Why this change?

Reply:
I found during debugging that the index returned during picking for thin instances is incorrect. Therefore, I believe that thin instances should not handle themselves, whereas instantiated models do have their own handling. So, I added a new _generateThinInstanceColorData method to differentiate between the handling logic for thin instances and instantiated models without disrupting the existing logic. This is why hasThinInstances does not increment the ID, but directly adds the thinInstanceCount. In the _generateThinInstanceColorData method, it ignores the existence of the model itself and directly processes the thin instances, ensuring the index is correct during picking.

My English is not very good, so I'm not sure if you understand what I'm saying.

Reply:
Do you have a playground for me to test? I understand what you mean but I'm not sure to understand why we need it

Reply:
in here: #JT0999#2，please see in host.

Reply:
@deltakosh Have you tested it?

Reply:
Ok this is great! My only ask would be to factorize the code between `_generateThinInstanceColorData` and `_generateColorData` as most of the code is similar

Reply:
I'm not sure how to factorize, so I modified the parameter list of the _generateThinInstanceColorData function. Make it look different from the _generaColorData function.